R/pipe_continuation_linter.R

```diff
@@ -0,0 +1,41 @@
+#' @describeIn linters check that all commas are followed by spaces, but do not
+#' have spaces before them.
+#' @importFrom xml2 xml_find_all as_list
+#' @export
+pipe_continuation_linter <- function(source_file) {
+  x <- global_xml_parsed_content(source_file)
+  if (is.null(x)) {
+    return()
+  }
+
+  # Find all expressions that span more than one line
+  # and contain specials named '%>%' that are on the same line
+
+  expressions_spanning_multiple_lines <- "expr[@line1 != @line2]"
+  pipes_on_same_line <- "SPECIAL[text() = '%>%' and @line1 = following::SPECIAL[text() = '%>%']/@line1]"
+
+  exprs <- xml_find_all(x,
```

Comment:
Not being familiar with Xpath, these statements look obscure, but powerful! Maybe in some future, all linters can reduce to a list of Xpath statements with an accompanying message.